src/Microsoft.Data/GroupBy.cs

```diff
@@ -64,21 +85,24 @@ public GroupBy(DataFrame dataFrame, int groupByColumnIndex, Dictionary<TKey, ICo
             _dataFrame = dataFrame ?? throw new ArgumentException(nameof(dataFrame));
         }
 
-        private delegate void ColumnDelegate(int columnIndex, long rowIndex, IEnumerable<long> rows, TKey key, bool firstGroup);
+        private delegate void ColumnDelegate(int columnIndex, long rowIndex, ICollection<long> rows, TKey key, bool firstGroup);
         private delegate void GroupByColumnDelegate(long rowNumber, TKey key);
-        private void EnumerateColumnsWithRows(GroupByColumnDelegate groupByColumnDelegate, ColumnDelegate columnDelegate)
+        private void EnumerateColumnsWithRows(GroupByColumnDelegate groupByColumnDelegate, ColumnDelegate columnDelegate, params string[] columnNames)
         {
             long rowNumber = 0;
             bool firstGroup = true;
             foreach (KeyValuePair<TKey, ICollection<long>> pairs in _keyToRowIndicesMap)
             {
                 groupByColumnDelegate(rowNumber, pairs.Key);
                 ICollection<long> rows = pairs.Value;
+                IEnumerable<string> columns = columnNames;
+                if (columnNames.Length == 0)
```

Comment:
> Won't this NullRef when you call it without any columnNames?

Nope. I had the same question so I explicitly debugged this. The compiler creates an Array<T>.Empty and passes that in to the method. I didn't investigate why this happens instead of passing in a null though. 

> Ensure we have a test for this scenario

We do. The existing tests in `TestGroupBy` cover this